src/value.rs

```diff
@@ -27,35 +29,144 @@ impl NumericValue {
     }
 }
 
-#[derive(PartialEq, Debug)]
+impl PartialOrd for NumericValue {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        match (self.0.as_i64(), self.0.as_u64(), self.0.as_f64()) {
```

Comment:
About the implementation. Do we want (need?) to compare all the possible combinations? I would really like to simplify this implementation...